doc/modules/neural_networks_supervised.rst

```diff
@@ -0,0 +1,222 @@
+below::
+
+.. figure:: ../auto_examples/neural_networks/images/plot_elm_scale_C_and_weight_scale_001.png
+   :target: ../auto_examples/neural_networks/plot_elm_scale_C_and_weight_scale.html
+   :align: center
+   :scale: 100%
+
+.. topic:: Examples:
+
+   * :ref:`example_neural_networks_plot_elm_scale_C_and_weight_scale.py`
+
+.. figure:: ../auto_examples/neural_networks/images/plot_elm_hyperparameters_001.png
+   :target: ../auto_examples/neural_networks/plot_elm_hyperparameters.html
+   :align: center
+   :scale: 100%
+
+.. topic:: Examples:
+
+   * :ref:`example_neural_networks_plot_elm_hyperparameters.py`
+
+Higher `n_hidden` provides ELM with a higher capacity to learn complex 
+functions. It allows ELM to have more feature detectors in the hidden 
+layer that characterize the training dataset. 
+
+This implements :ref:`classification <elm_classification>` for classification 
+and :ref:`regression <elm_regression>` for regression. They support both dense
+(``numpy.ndarray`` and convertible to that by ``numpy.asarray``) and
+sparse (any ``scipy.sparse``) sample vectors as input. They also support
+on-line and batch-based training.
+
+
+.. _elm_classification:
+
+Classification
+==============
+
+:class:`ELMClassifier` supports binary-class, multi-class, and
+multi-label classification on datasets.
+
+Like other classifiers, :class:`ELMClassifier` trains on a dataset using the
+fit method that accepts two input parameters: an array X of size 
+``(n_samples, n_features)``, which represents the training samples, and an array y 
+of integer values of size ``(n_samples)``, which holds the class labels respective
+to the training samples::
+
+    >>> from sklearn.neural_network import ELMClassifier
+    >>> X = [[0, 0], [1, 1]]
+    >>> y = [0, 1]
+    >>> clf = ELMClassifier(random_state=1)
+    >>> clf.fit(X, y)
+    ELMClassifier(C=1, activation='tanh', batch_size=None, class_weight=None,
+           n_hidden=500, random_state=1, verbose=False, warm_start=False,
+           weight_scale=1.0)
+
+After training, the model can predict the class of a new sample::
+
+    >>> clf.predict([[2., 2.]])
+    array([1])
+
+Weighted classification
+-----------------------
+
+For imbalanced datasets, giving more weight to classes having less samples than 
+others could help improve accuracy. 
+
+:class:`ELMClassifier` has the parameter ``class_weight`` in
+its ``fit`` method that can be set either to ``None`` - where no weight is
+given to any class; ``auto`` - where weight is given to a sample based
+on its class distribution in the dataset; and a dictionary of the form
+``{class_label : class_weight}``.
+
+Figure 2 shows the result of adding weight to the minority class, where the 
+decision boundary surrounding that class becomes larger in radius as more weight is given.
+
+.. figure:: ../auto_examples/neural_networks/images/plot_weighted_elm_001.png
+   :target: ../auto_examples/neural_networks/plot_weighted_elm.html
+   :align: center
+   :scale: 100%
```

Comment:
FYI this dataset here has been expertly crafted for sample weights: http://scikit-learn.org/stable/auto_examples/svm/plot_weighted_samples.html but feel free to keep this one.